docs(architecture): SCRUM-45 Document a modular monolith

adnlv opened the pull request:
**Jira**: [SCRUM-45](https://epam-team-php.atlassian.net/browse/SCRUM-45)

## Description

<!-- Short summary of what this PR does -->

Provide a modular monolith architecture documentation.

## How to roll back?

<!-- How to roll back these changes -->

Revert this pull request.

## How has this been tested?

<!-- Describe the tests that were run to verify your changes -->
<!-- Provide reproducible instructions -->

## Media attachments (optional)

<!-- Drag & drop images here or paste links -->

<img width="836" height="328" alt="Screenshot_20251211_111742" src="https://github.com/user-attachments/assets/e04e0b11-0980-4cbb-929b-d7dcc6cbe5bc" />

adnlv committed:
docs(SCRUM-45): Add modular monolith structure in detail

Adds a comprehensive guide on the modular monolith structure,
explaining its directory layout, responsibilities for each
module/component, and best practices. Aims to enhance maintainability
and facilitate future module extraction.

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+
+## Modular monolith structure
+
+The structure is solid and modular: each **feature module** (Iam, Profile, Tweet) follows the same vertical-slice
```
```suggestion
The structure is solid and modular: each **module** (Iam, Profile, Tweet) follows the same vertical-slice
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+## Modular monolith structure
+
+The structure is solid and modular: each **feature module** (Iam, Profile, Tweet) follows the same vertical-slice
+pattern (Resources / UI / Application / Domain / Infrastructure). That’s exactly what you want for a modular monolith:
```
```suggestion
pattern (UI / Application / Domain / Infrastructure). That’s exactly what you want for a modular monolith:
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+
+### `src/` (top level)
+
+**What:** root for all application code, organized by feature/module.
```
```suggestion
**What:** root for all application code, organized by modules.
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+### `src/` (top level)
+
+**What:** root for all application code, organized by feature/module.
+**Why:** encourages feature ownership, reduces cross-cutting coupling, simplifies CI/test scoping.
```
```suggestion
**Why:** encourages context ownership, reduces cross-cutting coupling, simplifies CI/test scoping.
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+├── Profile/
+│   └── ...
+└── Tweet/
+    └── ...
```
```suggestion
    └── ...
└── Shared/
    └── ...
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+**What:** root for all application code, organized by feature/module.
+**Why:** encourages feature ownership, reduces cross-cutting coupling, simplifies CI/test scoping.
+**Pitfalls / guidance:** keep a small `Shared` area for truly cross-cutting contracts (not business logic). Add
+top-level `tests/` and `migrations/` or document migration ownership per module.
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+**Pitfalls / guidance:** keep a small `Shared` area for truly cross-cutting contracts (not business logic). Add
+top-level `tests/` and `migrations/` or document migration ownership per module.
+
+### `src/Iam/resources/config/`
```
@adanyl0v Not the time for that. Remove it for now.

ihpr commented on docs/modular-monolith.md:
````diff
@@ -0,0 +1,233 @@
+```text 
+src/
+├── Iam/
````
```suggestion
├── IAM/
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │       ├── security.yaml
+│   │       └── routes.yaml
+│   ├── UI/
+│   │   └── Rest/
```
```suggestion
│   │   └── REST/
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │   ├── Validator/  # (interfaces) e.g., UniqueEmailCheckerInterface (define contract here) 
+│   │   ├── Handler/    # Validate DTO, call validators/interfaces, orchestrate use case
+│   │   ├── Command/
+│   │   ├── Dto/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │       ├── Controller/
+│   │       └── Exception/
+│   ├── Application/
+│   │   ├── Validator/  # (interfaces) e.g., UniqueEmailCheckerInterface (define contract here) 
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │       └── routes.yaml
+│   ├── UI/
+│   │   └── Rest/
+│   │       ├── Request/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │       └── Exception/
+│   ├── Application/
+│   │   ├── Validator/  # (interfaces) e.g., UniqueEmailCheckerInterface (define contract here) 
+│   │   ├── Handler/    # Validate DTO, call validators/interfaces, orchestrate use case
```
```suggestion
│   │   ├── UseCase/    # Stores handlers, queries, comamnds, DTO
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   ├── Application/
+│   │   ├── Validator/  # (interfaces) e.g., UniqueEmailCheckerInterface (define contract here) 
+│   │   ├── Handler/    # Validate DTO, call validators/interfaces, orchestrate use case
+│   │   ├── Command/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │   ├── Handler/    # Validate DTO, call validators/interfaces, orchestrate use case
+│   │   ├── Command/
+│   │   ├── Dto/
+│   │   ├── Exception/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │   ├── Command/
+│   │   ├── Dto/
+│   │   ├── Exception/
+│   │   └── Service/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │   └── Service/
+│   ├── Domain/
+│   │   ├── Model/
+│   │   ├── ValueObject/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   ├── Domain/
+│   │   ├── Model/
+│   │   ├── ValueObject/
+│   │   ├── Repository/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │   ├── Model/
+│   │   ├── ValueObject/
+│   │   ├── Repository/
+│   │   └── Exception/
```
```suggestion
```

ihpr commented on docs/modular-monolith.md:
```diff
@@ -0,0 +1,233 @@
+│   │   ├── Dto/
+│   │   ├── Exception/
+│   │   └── Service/
+│   ├── Domain/
```
@adanyl0v Domain -> Name -> Model -> (Entity, Aggregate, Value object, RepositoryInterafce). Examples:
- Domain -> User -> Model -> User.php
- Domain -> User -> Model -> Email.php
- Domain -> User -> Model -> UserRepository.php

ihpr reviewed (changes requested):
@adanyl0v The idea of a such detailed documentation is unclear to me. Why do we need to explain each class before its even created? It's an overhead. Decribe only main catalogs like UI, Application (with UseCase), Infrastructure, Domain. Give examples what can be stored inside.

Extend this doc while adding new patterns and folders to the aplication.

adnlv committed:
docs(architecture): SCRUM-45 Remove modular monolith guide section

adnlv committed:
docs(architecture): SCRUM-45 Revise modular monolith guide

Enhance guide with detailed structure and clarified component
responsibilities to improve maintainability. Remove overdetailed
explanations to simplify the documentation.
